core/cas-server-core-logging/src/main/java/org/apereo/cas/logging/web/ThreadContextMDCServletFilter.java

```diff
@@ -80,6 +81,14 @@ public void doFilter(final ServletRequest servletRequest, final ServletResponse
             Collections.list(request.getAttributeNames())
                     .forEach(a -> addContextAttribute(a, request.getAttribute(a)));
 
+            final Enumeration<String> requestHeaderNames = request.getHeaderNames();
```

Comment:
Excellent.

I recommend you simply clean up this block with the following:

```
Collections.list(request.getHeaderNames()).forEach(a -> addContextAttribute(a, request.getHeader(a)));
```

Which not only reduced LOC but keeps it consistent with the line above it. With the null-check of course. Otherwise, looks brilliant.

Reply:
The change has been submitted, hopefully the third time is the charm :). Thank you again for the feedback... I am primarily a Python and Perl dev so I am not overly familiar with Java one-liners and idiosyncrasies.

Reply:
You did perfectly fine. Thank you!